### PR TITLE
Make causal impact into market impact

### PR DESCRIPTION
**Pull request by hideaki**
### Description
Make causal impact into market impact, using code from MarketImpact.

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/causal_impact.R [hunk 1]
```diff
@@ -20,20 +20,27 @@ glance.bsts <- function(x) {
...
   time_col <- col_name(substitute(time))
-  do_causal_impact_(df, time_col, formula = formula, ...)
+  value_col <- col_name(substitute(value))
+  market_col <- col_name(substitute(market))
+  do_market_impact_(df, time_col, value_col, market_col, ...)
 }
 
 #' @param df - Data frame
 #' @param time_col - Column that has time data
 #' @param formula - Formula with target value column on the left-hand side, and predictor columns on the right-hand side. e.g. y ~ predictor1 + predictor2
```
Comment (double-y) [inline R/causal_impact.R]: Should we still keep this argument?

File: R/causal_impact.R [hunk 2]
```diff
@@ -20,20 +20,27 @@ glance.bsts <- function(x) {
...
-#' @param intervention_time - The point of time where intervention happened.
+#' @param event_time - The point of time where intervention happened.
```
Comment (double-y) [inline R/causal_impact.R]: "when event happened" might be better

File: R/causal_impact.R [hunk 3]
```diff
@@ -54,14 +61,20 @@ do_causal_impact <- function(df, time, formula, ...) {
...
+                              time_unit = "day", fun.aggregate = sum,
+                              formula = NULL, event_time = NULL, output_type = "series",
                               na_fill_type = "spline", na_fill_value = 0,
                               niter = NULL, standardize.data = NULL, prior.level.sd = NULL, nseasons = NULL, season.duration = NULL, dynamic.regression = NULL, ...) {
   validate_empty_data(df)
 
-  y_colname <- all.vars(lazyeval::f_lhs(formula))
-  predictor_column_names <- all.vars(lazyeval::f_rhs(formula))
-  all_column_names <- all.vars(formula)
+  if (!is.null(formula)) { # dead code for now.
```
Comment (double-y) [inline R/causal_impact.R]: Why don't you remove this for now?

File: R/MarketMatching.R [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+#' @import foreach
+#' @importFrom parallel detectCores
+#' @importFrom data.table rbindlist
+#' @import dplyr
+#' @import iterators
+#' @import utils
+#' @import dtw
+#' @importFrom doParallel registerDoParallel stopImplicitCluster
+#'
+#' @export best_matches
```
Comment (double-y) [inline R/MarketMatching.R]: Should we export this?

File: R/MarketMatching.R [hunk 5]
```diff
@@ -0,0 +1,665 @@
...
+  }
+
+  ### Return the results
+  object <- list(BestMatches=shortest_distances, Data=as.data.frame(saved_data), MarketID=id_variable, MatchingMetric=matching_variable, DateVariable=date_variable)
+  class(object) <- "matched_market"
+  return (object)
+}
+
+#' best_matches from zoo object version
+#' @export
```
Comment (double-y) [inline R/MarketMatching.R]: Should we export this?

File: R/MarketMatching.R [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+#' @param alpha Desired tail-area probability for posterior intervals. For example, 0.05 yields 0.95 intervals
+#' @param prior_level_sd Prior SD for the local level term (Gaussian random walk). Default is 0.01. The bigger this number is, the more wiggliness is allowed for the local level term.
+#' Note that more wiggly local level terms also translate into larger posterior intervals.
+#' @param control_matches Number of matching control markets to use in the analysis
+
+#' @importFrom scales comma
+#' @import ggplot2
+#' @import zoo
+
+#' @export inference
```
Comment (double-y) [inline R/MarketMatching.R]: Should we export this?

File: R/causal_impact.R (hunk 1 above)
Comment (hideaki) [inline R/causal_impact.R]: removed it.

File: R/causal_impact.R (hunk 2 above)
Comment (hideaki) [inline R/causal_impact.R]: updated the comment.

File: R/causal_impact.R (hunk 3 above)
Comment (hideaki) [inline R/causal_impact.R]: removed.

File: R/MarketMatching.R (hunk 5 above)
Comment (hideaki) [inline R/MarketMatching.R]: Removed this one and other unnecessary exports

File: R/MarketMatching.R (hunk 6 above)
Comment (hideaki) [inline R/MarketMatching.R]: removed export.

**Pull request merged**